### DISCUSSION TITLE
Alerting (alerts)

### DISCUSSION THREAD

**Discussion by TmLev** [grafana/tempo#2082] [Q&A] [answered]
Is it possible to generate alerts from Tempo traces?

My use case is sending alerts for every trace with `http.status_code = 500`. Can't find any relevant info for this in documentation.

Also, I tried to configure Tempo as a datasource for alerts, but it shows "Alerting not supported" when I choose Tempo in Datasources tab:

<img width="624" alt="image" src="https://user-images.githubusercontent.com/22966523/217309015-1d99b3cc-1ad7-4d66-9002-4b0d112784be.png">

I use Grafana v9.2.4 and latest Tempo.

**Comment by joe-elliott** [comment, marked as answer]
It is not possible to directly generate alerts from Tempo. However, you can use the metrics generator component to write metrics to a prometheus remote write compatible endpoint.

You could then build alerts based on those metrics which could include every 500 if you wanted. It is likely that this is a future feature we will add to Tempo, but it is currently unplanned.

**Reply by omri-shilton** [reply]
We have metrics-generator and the labels on the spanmetrics do not have labels of `http.status_code`.
Do you have a suggestion on how to solve this?
these are the only ones available:
<img width="236" alt="image" src="https://github.com/user-attachments/assets/131afd15-3b41-46ea-9c2d-2ffb7007af04" />

**Comment by wontonst** [comment]
has this feature been built out yet? alerting based on 500 status code rate seems like a pretty standard thing for a telemetry platform to have

**Reply by joe-elliott** [reply]
I do not know the details, but my impression is that with a feature flag Grafana will generate alerts on Tempo as a datasource. I'm struggling to find good docs, but I'm finding breadcrumbs like this closed issue:

https://github.com/grafana/grafana/issues/104289

I know we are offering a private preview of TraceQL alerting in cloud, but I know that's not quite what you're asking.

**Comment by knylander-grafana** [comment]
I found this page in the Grafana docs that describe how to do trace-based alerts: https://grafana.com/docs/grafana/latest/alerting/examples/trace-based-alerts/

Does this help?
